package/kedro_viz/integrations/kedro/sqlite_store.py

```diff
@@ -60,10 +84,112 @@ def to_json(self) -> str:
         return json.dumps(session_dict)
 
     def save(self):
-        """Save the session store info on db ."""
+        """Save the session store info on db and uploads it to the cloud if a remote cloud path is provided ."""
         engine, session_class = create_db_engine(self.location)
         Base.metadata.create_all(bind=engine)
         database = next(get_db(session_class))
-        session_store_data = RunModel(id=self._session_id, blob=self.to_json())
+
+        session_store_data = RunModel(id=self._session_id, blob=self._to_json())
         database.add(session_store_data)
         database.commit()
+        self.sync()
+
+    def _upload(self):
+        """Uploads the session store database file to the specified remote path on the cloud storage."""
+        # Connect to the remote file system
+        protocol, _ = get_protocol_and_path(self._remote_path)
+        db_name = _get_dbname()
+        fs = fsspec.filesystem(protocol)
+        # Open the local database file and upload it to the remote path
+        with open(self.location, "rb") as file:
+            with fs.open(f"{self._remote_path}/{db_name}", "wb") as s3f:
+                s3f.write(file.read())
+
+    def _download(self) -> List[str]:
+        """Downloads all the session store database files from the specified remote path on the cloud storage
+        to your local project.
+        Note: All the database files are deleted after they are merged to the main session_store.db.
+
+        Returns:
+        A list of local filepath in string format for all the databases
+
+        """
+
+        # Connect to the remote file system
+        protocol, _ = get_protocol_and_path(self._remote_path)
+        fs = fsspec.filesystem(protocol)
+        # Find all the databases at the remote path
+        databases = fs.glob(f"{self._remote_path}/*.db")
+        databases_location = []
+        # Download each database to a local filepath
+        for database in databases:
+            database_name = Path(database).name
+            with fs.open(database, "rb") as file:
+                db_data = file.read()
+                db_loc = f"{self._path}/{database_name}"
```

Comment:
This `db_loc` won't work well as a string because there's a `/` in it. You should  use `Pathlib` instead for manipulating filepaths 🙂